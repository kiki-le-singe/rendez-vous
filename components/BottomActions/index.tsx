import React from "react";
import { StyleSheet, View } from "react-native";

import UnderlineText from "../UnderlineText";
import CopyIcon from "../../assets/svg/js/CopyIcon";
import CutIcon from "../../assets/svg/js/CutIcon";
import BinIcon from "../../assets/svg/js/BinIcon";
import ActionsLinks from "../ActionsLinks";
import Button from "../Button";
import Divider from "../Divider";

export default function BottomActions() {
  const [isChoose, setIsChoose] = React.useState(false);
  const [isCome, setIsCome] = React.useState(false);
  const [isNotCome, setIsNotCome] = React.useState(false);

  const chooseText = "Choisi(e)";
  const comeText = "Venu";
  const notComeText = "Pas venu";

  function handleChoose() {
    setIsChoose(!isChoose);
  }
  function handleCome() {
    setIsCome(!isCome);
    isNotCome && setIsNotCome(false);
  }
  function handleNotCome() {
    setIsNotCome(!isNotCome);
    isCome && setIsCome(false);
  }

  return (
    <View>
      <View style={styles.left}>
        {isChoose ? (
          <Button onPress={handleChoose} label={chooseText} theme="green" />
        ) : (
          <UnderlineText
            onPress={handleChoose}
            text={chooseText}
            disabled={false}
          />
        )}

        {isCome ? (
          <Button onPress={handleCome} label={comeText} theme="darkBlue" />
        ) : (
          <UnderlineText
            onPress={handleCome}
            text={comeText}
            disabled={false}
          />
        )}

        {isNotCome ? (
          <Button onPress={handleNotCome} label={notComeText} theme="red" />
        ) : (
          <UnderlineText
            onPress={handleNotCome}
            text={notComeText}
            disabled={false}
          />
        )}
      </View>

      <Divider />

      <ActionsLinks containerStyles={styles.actionsLinks}>
        <UnderlineText text="Copier" icon={<CopyIcon />} />
        <UnderlineText text="Couper" icon={<CutIcon />} />
        <UnderlineText text="Supprimer" icon={<BinIcon />} />
      </ActionsLinks>

      <Divider />

      <View style={styles.right}>
        <Button
          stylesContainer={[styles.button, { flex: 1 }]}
          label="Plus d'options (produits, remises, ...)"
        />
        <Button
          stylesContainer={[styles.button, { maxWidth: 180 }]}
          label="Encaisser 230,00 â‚¬"
          theme="green"
        />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  left: {
    flexDirection: "row",
    justifyContent: "center",
    gap: 16,
  },
  right: {
    gap: 16,
  },
  button: {
    alignSelf: "center",
    height: 56,
  },
  actionsLinks: {
    justifyContent: "center",
  },
});
