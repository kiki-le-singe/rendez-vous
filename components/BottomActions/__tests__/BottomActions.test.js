import * as React from "react";
import renderer from "react-test-renderer";

import BottomActions from "..";

it(`renders correctly`, () => {
  const tree = renderer.create(<BottomActions />).toJSON();

  expect(tree).toMatchSnapshot();
});
