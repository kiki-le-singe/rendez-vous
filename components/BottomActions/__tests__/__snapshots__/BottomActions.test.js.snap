// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      {
        "flexDirection": "row",
        "gap": 16,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 6,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 13,
              "fontWeight": "600",
              "textDecorationLine": "underline",
            },
            {
              "color": "#34423E",
            },
          ]
        }
      >
        Choisi(e)
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 6,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 13,
              "fontWeight": "600",
              "textDecorationLine": "underline",
            },
            {
              "color": "#34423E",
            },
          ]
        }
      >
        Venu
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 6,
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 13,
              "fontWeight": "600",
              "textDecorationLine": "underline",
            },
            {
              "color": "#34423E",
            },
          ]
        }
      >
        Pas venu
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "borderTopColor": "#ECEEED",
        "borderTopWidth": 1,
        "marginVertical": 22,
      }
    }
  />
  <View
    style={
      [
        {
          "flexDirection": "row",
          "gap": 18,
        },
        {
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 6,
          "opacity": 1,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="14"
        fill="none"
        focusable={false}
        height="16"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 16,
              "width": 14,
            },
          ]
        }
        vbHeight={16}
        vbWidth={14}
        width="14"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M10 0.666687H2.00001C1.26667 0.666687 0.666672 1.26669 0.666672 2.00002V11.3334H2.00001V2.00002H10V0.666687ZM12 3.33335H4.66667C3.93334 3.33335 3.33334 3.93335 3.33334 4.66669V14C3.33334 14.7334 3.93334 15.3334 4.66667 15.3334H12C12.7333 15.3334 13.3333 14.7334 13.3333 14V4.66669C13.3333 3.93335 12.7333 3.33335 12 3.33335ZM12 14H4.66667V4.66669H12V14Z"
            fill={
              {
                "payload": 4284444778,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          [
            {
              "fontSize": 13,
              "fontWeight": "600",
              "textDecorationLine": "underline",
            },
            {
              "color": "#34423E",
            },
          ]
        }
      >
        Copier
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 6,
          "opacity": 1,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="14"
        bbWidth="14"
        fill="none"
        focusable={false}
        height="14"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 14,
              "width": 14,
            },
          ]
        }
        vbHeight={14}
        vbWidth={14}
        width="14"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M5.42666 4.09325C5.58 3.75992 5.66666 3.39325 5.66666 2.99992C5.66666 1.52659 4.47333 0.333252 2.99999 0.333252C1.52666 0.333252 0.333328 1.52659 0.333328 2.99992C0.333328 4.47325 1.52666 5.66659 2.99999 5.66659C3.39333 5.66659 3.75999 5.57992 4.09333 5.42659L5.66666 6.99992L4.09333 8.57325C3.75999 8.41992 3.39333 8.33325 2.99999 8.33325C1.52666 8.33325 0.333328 9.52659 0.333328 10.9999C0.333328 12.4733 1.52666 13.6666 2.99999 13.6666C4.47333 13.6666 5.66666 12.4733 5.66666 10.9999C5.66666 10.6066 5.58 10.2399 5.42666 9.90659L7 8.33325L11.6667 12.9999H13.6667V12.3333L5.42666 4.09325ZM2.99999 4.33325C2.26666 4.33325 1.66666 3.73992 1.66666 2.99992C1.66666 2.25992 2.26666 1.66659 2.99999 1.66659C3.73333 1.66659 4.33333 2.25992 4.33333 2.99992C4.33333 3.73992 3.73333 4.33325 2.99999 4.33325ZM2.99999 12.3333C2.26666 12.3333 1.66666 11.7399 1.66666 10.9999C1.66666 10.2599 2.26666 9.66659 2.99999 9.66659C3.73333 9.66659 4.33333 10.2599 4.33333 10.9999C4.33333 11.7399 3.73333 12.3333 2.99999 12.3333ZM7 7.33325C6.81333 7.33325 6.66666 7.18659 6.66666 6.99992C6.66666 6.81325 6.81333 6.66659 7 6.66659C7.18666 6.66659 7.33333 6.81325 7.33333 6.99992C7.33333 7.18659 7.18666 7.33325 7 7.33325ZM11.6667 0.999919L7.66666 4.99992L9 6.33325L13.6667 1.66659V0.999919H11.6667Z"
            fill={
              {
                "payload": 4284444778,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          [
            {
              "fontSize": 13,
              "fontWeight": "600",
              "textDecorationLine": "underline",
            },
            {
              "color": "#34423E",
            },
          ]
        }
      >
        Couper
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 6,
          "opacity": 1,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="12"
        fill="none"
        focusable={false}
        height="16"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 16,
              "width": 12,
            },
          ]
        }
        vbHeight={16}
        vbWidth={12}
        width="12"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M9.33332 5.5V13.8333H2.66666V5.5H9.33332ZM8.08332 0.5H3.91666L3.08332 1.33333H0.166656V3H11.8333V1.33333H8.91666L8.08332 0.5ZM11 3.83333H0.99999V13.8333C0.99999 14.75 1.74999 15.5 2.66666 15.5H9.33332C10.25 15.5 11 14.75 11 13.8333V3.83333Z"
            fill={
              {
                "payload": 4281614910,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          [
            {
              "fontSize": 13,
              "fontWeight": "600",
              "textDecorationLine": "underline",
            },
            {
              "color": "#34423E",
            },
          ]
        }
      >
        Supprimer
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "borderTopColor": "#ECEEED",
        "borderTopWidth": 1,
        "marginVertical": 22,
      }
    }
  />
  <View
    style={
      {
        "gap": 16,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        animatedStyle={
          {
            "value": {},
          }
        }
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#34423E",
            "borderRadius": 6,
            "borderWidth": 1,
            "flex": 1,
            "height": 56,
            "justifyContent": "center",
            "padding": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "#34423E",
            }
          }
        >
          Plus d'options (produits, remises, ...)
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        animatedStyle={
          {
            "value": {},
          }
        }
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#48BB78",
            "borderRadius": 6,
            "height": 56,
            "justifyContent": "center",
            "maxWidth": 180,
            "padding": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "#FFFFFF",
            }
          }
        >
          Encaisser 230,00 €
        </Text>
      </View>
    </View>
  </View>
</View>
`;
